File: 프로그래머스/JavaScript/Lv. 0/181932. 코드 처리하기/코드 처리하기.js
function solution(code) {
  let mode = 0;
  let answer = '';

  for (let idx = 0; idx < code.length; idx++) {
    const char = code[idx];

    if (char === '1') {
      mode = 1 - mode;
    } else if (idx % 2 === mode) {
      answer += char;
    }
  }

  return answer || 'EMPTY';
}

File: 프로그래머스/Java/Lv. 0/181932. 코드 처리하기/코드 처리하기.java
class Solution {
	public String solution(String code) {
		StringBuilder sb = new StringBuilder();
		int mode = 0;

		for (int i = 0; i < code.length(); i++) {
			char c = code.charAt(i);

			if (c == '1') {
				mode = (mode == 0) ? 1 : 0;
				continue;
			}

			if (mode == 0) {
				if (i % 2 == 0) {
					sb.append(c);
				}
			} else {
				if (i % 2 == 1) {
					sb.append(c);
				}
			}
		}

		return sb.length() == 0 ? "EMPTY" : sb.toString();
	}
}